### PR TITLE
#782: Unify test logging style

### PR DESCRIPTION
**Pull request by benedeki**
* Using pom.xml properties to configure tests logging
* scalastyle and resources in parent pom.xml
* scalastyle to run only on sub-modules
* dao/pom.xml duplicated plugins removed

### CODE REVIEW

File: examples/src/test/resources/log4j.properties [hunk 1]
```diff
@@ -14,23 +14,34 @@
...
+# Suppress a spamming warning from CacheManager
+log4j.appender.forcachemanager=org.apache.log4j.ConsoleAppender
+log4j.appender.forcachemanager.target=System.err
+log4j.appender.forcachemanager.layout=org.apache.log4j.PatternLayout
+log4j.appender.forcachemanager.layout.ConversionPattern=@log.pattern@
+log4j.appender.forcachemanager.filter.01=org.apache.log4j.varia.StringMatchFilter
+log4j.appender.forcachemanager.filter.01.StringToMatch=Asked to cache already cached data.
+log4j.appender.forcachemanager.filter.01.AcceptOnMatch=@log.specialfilters.acceptonmatch@
+log4j.logger.org.apache.spark.sql.execution.CacheManager=@log.level.base@, forcachemanager
+log4j.additivity.org.apache.spark.sql.execution.CacheManager=false
```
Comment (Zejnilovic) [inline examples/src/test/resources/log4j.properties]: missing newline

File: menas/src/test/resources/log4j.properties [hunk 2]
```diff
@@ -14,11 +14,11 @@
...
 
 log4j.appender.general=org.apache.log4j.ConsoleAppender
 log4j.appender.general.target=System.err
 log4j.appender.general.layout=org.apache.log4j.PatternLayout
-log4j.appender.general.layout.ConversionPattern=[%p] %d{yy/MM/dd HH:mm:ss} %c{1}: %m%n
+log4j.appender.general.layout.ConversionPattern=@log.pattern@
 
-log4j.logger.za.co.absa.enceladus.menas.services.RunService=ERROR
\ No newline at end of file
+log4j.logger.za.co.absa.enceladus.menas.services.RunService=@log.level.testcode@
```
Comment (Zejnilovic) [inline menas/src/test/resources/log4j.properties]: missing newline

File: pom.xml [hunk 3]
```diff
@@ -181,7 +181,13 @@
         <cronstrue.version>1.79.0</cronstrue.version>
         <!--other properties-->
         <skip.integration.tests>true</skip.integration.tests>
-        <scalastyle.configLocation>${project.basedir}/scalastyle-config.xml</scalastyle.configLocation>
+        <!--suppress UnresolvedMavenProperty -->
```
Comment (Zejnilovic) [inline pom.xml]: This seems dangerous. Is this because of the sub-module poms?

Comment (benedeki) [reply pom.xml]: Added during development, missed it to remove. Will delete it.

File: conformance/pom.xml [hunk 4]
```diff
@@ -99,9 +99,11 @@
         </dependency>
     </dependencies>
     <build>
-        <sourceDirectory>src/main/scala</sourceDirectory>
```
Comment (lokm01) [inline conformance/pom.xml]: Why?

File: conformance/pom.xml [hunk 5]
```diff
@@ -177,17 +179,5 @@
                 </executions>
             </plugin>
         </plugins>
-        <resources>
```
Comment (lokm01) [inline conformance/pom.xml]: As long as filtering is done elsewhere...

File: dao/pom.xml [hunk 6]
```diff
@@ -64,53 +64,19 @@
             <artifactId>httpclient</artifactId>
             <version>4.5.8</version>
         </dependency>
-        <dependency>
```
Comment (lokm01) [inline dao/pom.xml]: Has this been ran on a proper hadoop environment and tested with oozie?

File: conformance/pom.xml (hunk 4 above)
Comment (benedeki) [reply conformance/pom.xml]: Moved to main `pom.xml`, as it was repeated in all sub-modules

File: conformance/pom.xml (hunk 5 above)
Comment (benedeki) [reply conformance/pom.xml]: Yeah, again done in the master `pom.xml`.
Was repeated in more modules, Furthermore now needed for test resources recognition and the variables expansion in `..../test/resources/log4j.properties`

File: dao/pom.xml (hunk 6 above)
Comment (benedeki) [reply dao/pom.xml]: The very same plugins were included in the master `'pom.xml`.  _Idea_ was pointing it out.
Compared and kept the main one. Bot no extra tests other then the ones included in the build has been done so far. What is the worry?

Comment (lokm01) [reply dao/pom.xml]: Worried about all sorts of binary incompatibilities that these were trying to solve.

Though it's possible that these were no longer needed.

**Comment by benedeki** [comment]
> 
> 
> To ensure everything still works we can:
> 
>     * Deploy this branch to Asgard
> 
>     * Run Standardization and Conformance Jobs
> 
>     * Schedule an Oozie job and make sure it works.
>       I'm not sure how else we can be comfortable that these changes do not break anything.

Standardization verified to work. With Oozie scheduling I would like to ask for assistance either @Zejnilovic or @lokm01

**Pull request merged**